app/src/main/java/ch/epfl/sweng/erpa/services/RemoteServicesProvider.java

```diff
@@ -2,11 +2,12 @@
 
 import com.annimon.stream.Optional;
 
+import ch.epfl.sweng.erpa.model.Game;
 import ch.epfl.sweng.erpa.model.UserProfile;
 
 public interface RemoteServicesProvider {
     // Here be data proxy
-    GameService getGameService();
+    DataService<Game> getGameService();
```

Comment:
We need a GameService Class here!
DataService<Game> is a repository class (e.g. it handles persistance only) whilst GameService should do data validation and possibly other business logic, Please user GameService as return type!